### DISCUSSION TITLE
Home Assistant OS 12.4

### DISCUSSION THREAD

**Discussion by sairon** [home-assistant/operating-system#3423] [Announcements]
## Known Issues

:warning: **Some Raspberry Pi users have reported system lockups with some Z-Wave and Zigbee USB adapters. This is an unresolved bug in the Raspberry Pi kernel, workaround is to use the DWC2 driver by adding `dtoverlay=dwc2` to `config.txt`. See #3362 for details.**

:warning: **Fujitsu Esprimo Q920 fails to boot with GRUB bootloader distributed in this OS release, updating to 12.4.rc1 may cause a boot failure and manual intervention might be needed, see #3348 for details.**

:warning: **For ODROID-N2 (and Home Assistant Blue) this release reverts U-Boot update which is believed to be the cause of some intermittent issues when booting from eMMC. If you had any trouble previously after an update to OS 11.5 or newer, please see #3227 and help us with tracing this issue. Also note that some new eMMC modules with the Kingston chips are not supported yet - this is tracked in #3250.**

## Home Assistant Operating System

* Bump Docker to v26.1.4 (#3411) @sairon
* Update Buildroot to 2024.02.3 (#3406) @sairon

## Build System

* Use Genimage for declarative image layout (#3388) @sairon
* Bump docker/build-push-action from 5.3.0 to 5.4.0 (#3405) @dependabot
* Bump docker/login-action from 3.1.0 to 3.2.0 (#3399) @dependabot

## Raspberry Pi

* Set tty0 in cmdline as the default preferred console (#3402) @sairon
* RaspberryPi: Update kernel to 6.6.31 - stable_20240529 (#3400) @sairon

## Home Assistant Yellow

* Set tty0 in cmdline as the default preferred console (#3402) @sairon
* RaspberryPi: Update kernel to 6.6.31 - stable_20240529 (#3400) @sairon
* Linux: Update kernel to 6.6.32 (#3394) @sairon

## Home Assistant Green

* Linux: Update kernel to 6.6.33 (#3415) @sairon
* Fix writing to eMMC in U-Boot on Green (#3412) @sairon
* Set tty0 in cmdline as the default preferred console (#3402) @sairon
* Use U-Boot tarball for all boards (#3361) @sairon

## Open Virtual Appliance

* Linux: Update kernel to 6.6.33 (#3415) @sairon
* Set tty0 in cmdline as the default preferred console (#3402) @sairon
* Bump buildroot to update openvmtools to 12.3.0 (again) (#3370) @sairon

## Generic x86-64

* Linux: Update kernel to 6.6.33 (#3415) @sairon
* Set tty0 in cmdline as the default preferred console (#3402) @sairon

## Hardkernel ODROID

* Linux: Update kernel to 6.6.33 (#3415) @sairon
* Temporary revert ODROID-N2 U-Boot to 2023.01 to fix eMMC issues (#3409) @sairon
* Set tty0 in cmdline as the default preferred console (#3402) @sairon
* Use U-Boot tarball for all boards (#3361) @sairon

## ASUS Tinker

* Linux: Update kernel to 6.6.33 (#3415) @sairon
* Set tty0 in cmdline as the default preferred console (#3402) @sairon

## Khadas VIM Series

* Linux: Update kernel to 6.6.33 (#3415) @sairon
* Set tty0 in cmdline as the default preferred console (#3402) @sairon

## Generic aarch64

* Linux: Update kernel to 6.6.33 (#3415) @sairon
* Set tty0 in cmdline as the default preferred console (#3402) @sairon

<hr /><em>This discussion was created from the release <a href='https://github.com/home-assistant/operating-system/releases/tag/12.4'>Home Assistant OS 12.4</a>.</em>

**Comment by stefanuytterhoeven** [comment]
Ok, no update for me then...
Releasing a buggy kernel? Why don't you wait until it's solved?

**Reply by sairon** [reply]
Nothing changed since the previous release in this regard - we only give a heads-up here and suggest a way to fix it (which we don't want to apply automatically for couple of reasons). Also it doesn't affect everyone and resolves bunch of different issues.

**Reply by stefanuytterhoeven** [reply]
it's a "known issue" for release 12.4.  It looks like we just have to find out if it affects us or not. 
so, i'll wait. I don't feel the need to gamble...

**Comment by FloatingBoater** [comment]
Hi,
HAOS 12.4 on a stock Yellow fails to create any USB devices (e.g. Aeotec G5 coordinator, generic Realtek Bluetooth 5.1 Radio) leaving only `/dev/ttyAMA*`:
https://community.home-assistant.io/t/haos-12-4-borks-usb-devices-on-yellow-rpi-usb-kernel-bug/740946

Can I suggest a stronger wording in the release notes please?

Best regards,

James

**Reply by sairon** [reply]
It's a different issue than the one in release notes - Yellow already uses the DWC2 driver but we saw some problems on Yellow to, see #3347. It is intermittent and another reboot usually fixes it. Unfortunately we don't have a reliable way of reproducing it yet so it's hard to debug it further.

**Comment by peterjuras** [comment]
No way I'm updating to this release. It seems like everyone with a Raspberry Pi & Zigbee dongle is facing issues.

**Reply by sairon** [reply]
No, not everyone is facing issues. The issues were reported already with the previous release (moreover, the Zigbee ones are rather anectodal at this point) and there are literally thousands of people running it without problems, in this release we just gave a heads-up in the release notes and suggested a way of fixing it.

**Reply by peterjuras** [reply]
A lot of users in related issues mentioned that downgrading to 12.2 helps to fix their issue, users with similar setups to mine (Raspberry 4 + Sonoff Dongle E).

It's still hard to understand for me on whether it will break my system and whether the issue is gone in a future version, therefore I'd rather stay on the current one.

**Reply by Czarnykowal** [reply]
yep it's reseting on me after update. Raspberry 4B and zigbee dongle, adding the "dtoverlay=dwc2" tweak to CONFIG.TXT fixed ot for now

**Comment by dtremere** [comment]


**Reply by FloatingBoater** [reply]
Hi Doug,
Can I suggest posting your request for help on https://community.home-assistant.io/ as GitHub is more a place to discuss specific development and technical issues than detailed end-user support?
My initial suggestion is backup, wipe, reinstall HAOS, restore - happy to help more, just in a better context.
All the best,

James

**Comment by dtremere** [comment]
Sorry about that, I followed the link for the release notes and since it
seems to be related to the SLOT A / B design I thought it was the right
place. I have searched quite a bit to try and understand the A / B
situation better but can only find general descriptions of why it was
designed that way.
Anyhow, I'll remove my post and report in the community.
Thanks for the heads-up.
-Doug

**Comment by Baxxy13** [comment]
A freshly flashed `haos_rpi5-64-12.4.img.xz` isn't booting on Pi5B.
See: #3437 
An update from 12.3 to 12.4, on the other hand, works without any problems on my Pi5B-8GB.

**Comment by ThePatricide** [comment]
@sairon it seems the Docker engine version is affected by a severe vulnerability: https://www.docker.com/blog/docker-security-advisory-docker-engine-authz-plugin/
Is this an issue for Home Assistant and does it need to be patched quickly?

**Reply by sairon** [reply]
No, Docker on HAOS doesn't use any authorization plugins so it's not affected.
